### PR TITLE
Data acquisition from time-of-flight sensor

### PR DESCRIPTION
**Pull request by gunturiCM**
# This is a draft. The work is incomplete. Not to be merged.

## Main Changes
Implementation of a PD controller-based altitude hold mechanism written in Python and to be run on Raspberry Pi. The pressure sensor and time of flight sensors are to be integrated to R-Pi and based on altitude readings, R-Pi modifies the throttle, and attitude data from the radio with its own calculations and then sends it to ESP WROOM 32, where the basic attitude controller is running.

## Note: 
The simulations are coming soon.

### CODE REVIEW

**Comment by alphaville** [comment]
@gunturiCM I'm not sure whether you plan to carry out the entire development related to the altitude hold controller (and loitering?) in one branch and PR, but it's not a good idea. A brach/PR should be **atomic** and address one issue.

**Comment by gunturiCM** [comment]
> @gunturiCM I'm not sure whether you plan to carry out the entire development related to the altitude hold controller (and loitering?) in one branch and PR, but it's not a good idea. A brach/PR should be **atomic** and address one issue.

Yes! Professor, except for altitude hold-related files, the rest are there for my reference and only send dummy values. I do not intend to push all of the code into this single branch and as you suggested the PRs will be atomic. When complete this PR will only have the altitude hold controller-related scripts. Just for the time being I would like to have the other files in here.

**Comment by alphaville** [comment]
@gunturiCM I'm not sure what the difference between this PR and #79 is. It seems that 62 files have been changed, some of which are irrelevant to the altitude hold controller. I don't think we can merge this.

**Comment by gunturiCM** [comment]
> @gunturiCM I'm not sure what the difference between this PR and #79 is. It seems that 62 files have been changed, some of which are irrelevant to the altitude hold controller. I don't think we can merge this.

Yes, professor. This branch needs some cleaning before merging.  #79 has the implementation of an LQR controller for altitude hold, whereas this is an old branch that was developed under the idea of using a PID/PD controller for the same purpose. We have decided to use LQR at a later time and hence created a new PR #79.

File: raspberry/main.py.save
Comment (alphaville) [inline raspberry/main.py.save:1]: This `.save` file should not have been committed. Let us remove this file and update our `.gitignore`.

**Comment by alphaville** [comment]
@gunturiCM This PR involves 48 files, so we won't be able to merge it

File: raspberry/main.py.save
Comment (gunturiCM) [reply raspberry/main.py.save:1]: Creating an Issue.

**Comment by gunturiCM** [comment]
Closing this branch because it is no longer needed. A new control and estimation architecture for altitude hold has been successfully implemented.

**Pull request closed without merging**